test/python/circuit/test_library.py

```diff
@@ -144,34 +144,41 @@ def test_permutation_bad(self):
         self.assertRaises(CircuitError, Permutation, 4, [1, 0, -1, 2])
 
 
+@ddt
 class TestHiddenLinearFunctionLibrary(QiskitTestCase):
     """Test library of Hidden Linear Function circuits."""
+    def __q(self, x, hidden_function):
+        return np.dot(x.transpose(), np.dot(hidden_function, x))[0][0]
```

Comment:
Since this is just a one-liner, could you move this into `self.assertHLFIsCorrect`? Then the matrix for the hidden function is already defined locally so it is not needed as argument for `q` anymore